app/grandchallenge/cases/image_builders/nifti.py

```diff
@@ -0,0 +1,61 @@
+from pathlib import Path
+from typing import Set
+
+import SimpleITK
+
+from grandchallenge.cases.image_builders.types import ImageBuilderResult
+from grandchallenge.cases.image_builders.utils import convert_itk_to_internal
+
+
+def format_error(message):
+    return f"NifTI image builder: {message}"
+
+
+def image_builder_nifti(*, files: Set[Path], **_) -> ImageBuilderResult:
+    """
+    Constructs image objects from files in NifTI format (nii/nii.gz)
+
+    Parameters
+    ----------
+    files
+        Path to images that were uploaded during an upload session.
+
+    Returns
+    -------
+    An `ImageBuilder` object consisting of:
+     - a list of filenames for all files consumed by the image builder
+     - a list of detected images
+     - a list files associated with the detected images
+     - path->error message map describing what is wrong with a given file
+    """
+    errors = {}
+    new_images = set()
+    new_image_files = set()
+    consumed_files = set()
+    for file in files:
+        if not (file.name.endswith(".nii") or file.name.endswith(".nii.gz")):
+            continue
+
+        try:
+            img: SimpleITK.Image = SimpleITK.ReadImage(str(file.absolute()))
+        except RuntimeError:
```

Comment:
All of the exceptions that `ReadImage` can raise should be handled here. Is `RuntimeError` the only one that is possible?

Reply:
As far as I know that's the only type of exception raised by SimpleITK. The SimpleITK documentation is not very clear about this unfortunately.

Reply:
Lets see how it goes, the downside is that one bad nifti file could mean that the whole import is cancelled in which case the user will get an "unexpected error".